plugins/webp-uploads/hooks.php

```diff
@@ -681,38 +681,52 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
  */
 function webp_uploads_wepb_fallback() {
 	// Get mime type transforms for the site.
-	$transforms = webp_uploads_get_upload_image_mime_transforms();
+	$transforms   = webp_uploads_get_upload_image_mime_transforms();
+	$image_format = webp_uploads_get_image_output_format();
 
-	// We need to add fallback only if jpeg alternatives for the webp images are enabled for the server.
-	$preserve_jpegs_for_jpeg_transforms = isset( $transforms['image/jpeg'] ) && in_array( 'image/jpeg', $transforms['image/jpeg'], true ) && in_array( 'image/webp', $transforms['image/jpeg'], true );
-	$preserve_jpegs_for_webp_transforms = isset( $transforms['image/webp'] ) && in_array( 'image/jpeg', $transforms['image/webp'], true ) && in_array( 'image/webp', $transforms['image/webp'], true );
-	if ( ! $preserve_jpegs_for_jpeg_transforms && ! $preserve_jpegs_for_webp_transforms ) {
+	// We need to add fallback only if jpeg alternatives for the image_format images are enabled for the server.
+	$preserve_jpegs_for_jpeg_transforms       = isset( $transforms['image/jpeg'] ) && in_array( 'image/jpeg', $transforms['image/jpeg'], true ) && in_array( 'image/' . $image_format, $transforms['image/jpeg'], true );
+	$preserve_jpegs_for_image_type_transforms = isset( $transforms[ 'image/' . $image_format ] ) && in_array( 'image/jpeg', $transforms[ 'image/' . $image_format ], true ) && in_array( 'image/' . $image_format, $transforms[ 'image/' . $image_format ], true );
+	if ( ! $preserve_jpegs_for_jpeg_transforms && ! $preserve_jpegs_for_image_type_transforms ) {
 		return;
 	}
 
-	ob_start();
+	$detection_string = '';
+	// The fallback script can only handle a single image format at a time.
+	if ( 'webp' === $image_format ) {
+		$detection_string = 'data:image/webp;base64,UklGRh4AAABXRUJQVlA4TBEAAAAvAQAAAAfQ//73v/+BiOh/AAA=';
+	} elseif ( 'avif' === $image_format ) {
+		$detection_string = 'data:image/avif;base64,AAAAIGZ0eXBhdmlmAAAAAGF2aWZtaWYxbWlhZk1BMUIAAADybWV0YQAAAAAAAAAoaGRscgAAAAAAAAAAcGljdAAAAAAAAAAAAAAAAGxpYmF2aWYAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAABGgAAAB0AAAAoaWluZgAAAAAAAQAAABppbmZlAgAAAAABAABhdjAxQ29sb3IAAAAAamlwcnAAAABLaXBjbwAAABRpc3BlAAAAAAAAAAIAAAACAAAAEHBpeGkAAAAAAwgICAAAAAxhdjFDgQ0MAAAAABNjb2xybmNseAACAAIAAYAAAAAXaXBtYQAAAAAAAAABAAEEAQKDBAAAACVtZGF0EgAKCBgANogQEAwgMg8f8D///8WfhwB8+ErK42A=';
+	}
+	if ( '' === $detection_string ) {
+		return;
+	}
 
-	?>
-	( function( d, i, s, p ) {
+	$js_function = <<<JS
+	function webpUploadsDetectFallback( d, i, s, p, fallbackSrc, webpUploadsOutputFormat ) {
 		s = d.createElement( s );
-		s.src = '<?php echo esc_url_raw( plugins_url( '/fallback.js', __FILE__ ) ); ?>';
+		s.src = fallbackSrc;
 
 		i = d.createElement( i );
-		i.src = p + 'jIAAABXRUJQVlA4ICYAAACyAgCdASoCAAEALmk0mk0iIiIiIgBoSygABc6zbAAA/v56QAAAAA==';
+		i.src = p;
 		i.onload = function() {
 			i.onload = undefined;
-			i.src = p + 'h4AAABXRUJQVlA4TBEAAAAvAQAAAAfQ//73v/+BiOh/AAA=';
 		};
 
 		i.onerror = function() {
 			d.body.appendChild( s );
```

Comment:
It looks like `webpUploadsOutputFormat` isn't yet set as a global variable at this point. This would fix that:

```suggestion
			window.webpUploadsOutputFormat = webpUploadsOutputFormat;
			d.body.appendChild( s );
```

But perhaps it should be set as a data attribute on the `script` tag the same way that was done for `data-rest-api`?

Reply:
Good suggestion, I haven't quite gotten to testing this part yet.

Reply:
Actually, this isn't a global, it is passed as the 6th argument to the `webpUploadsDetectFallback` function. I guess we can document that function.

Reply:
Right, but it is passed into that function but then it isn't passed on anywhere from that. How does it get passed to `wpPerfLab.webpUploadsFallbackWebpImages()` in `fallback.js`?

Reply:
trying that..

Reply:
addressed in [`d3dd80a` (#1176)](https://github.com/WordPress/performance/pull/1176/commits/d3dd80aa10549672b77d99d059ceed321fb4c203)